hub/core/chunk_engine/write.py

```diff
@@ -14,17 +14,32 @@
 from .read import read_index_map, read_tensor_meta, tensor_exists
 
 
+def tuple_to_list(shape: Tuple):
+    shapeArray = np.asarray(shape)
+    return shapeArray.tolist()
+
+
 def write_tensor_meta(key: str, storage: StorageProvider, meta: dict):
-    storage[get_tensor_meta_key(key)] = pickle.dumps(meta)
+    meta["min_shape"] = tuple_to_list(meta["min_shape"])
+    meta["max_shape"] = tuple_to_list(meta["max_shape"])
+    storage[get_tensor_meta_key(key)] = bytes(json.dumps(meta), "utf-8")
 
 
 def write_index_map(key: str, storage: StorageProvider, index_map: list):
     index_map_key = get_index_map_key(key)
-    storage[index_map_key] = pickle.dumps(index_map)
+    storage[index_map_key] = bytes(json.dumps(index_map), "utf-8")
```

Comment:
instead of `write_index_map` and `read_index_map`, we should use a class that automatically handles this logic. then, when i go to add an entry, i could just do 

`index_map.add_entry(start_byte=something, end_byte=something, chunk_names=..., shape=...)`

this add_entry function should automatically encode.

the state of `index_map` would be a `bytearray`.

then when reading, i can do

`index_map[0].chunk_names`

these reads should automatically decode.

```python
class SampleTracker:  # TODO: rename
    def __init__(self):
        self.state = bytearray() # TODO: rename

    def add_entry(self, *kwargs): # TODO: define arguments STATICALLY not kwargs
        array = encode(kwargs)
        self.state += array.tobytes()

    def __getitem__(self, item):
        # TODO: get the item referenced from state
        # return dictionary (decode array)
       pass
```

Reply:
so your file would look like this:


one line = 1 encoded array (bytes)
```
3291571239045okjgsldkgjlagjlasdjflalsdkfj
214923598134iothjgkldfsbkh
12345i13745987134i6245ok7jk2l4
13245972134t98ugkeraghskdfhgls
431543ht52k34h6k2jh5y134
```

^ this file has 5 index entries